apps/opik-backend/src/main/java/com/comet/opik/infrastructure/llm/gemini/GeminiLlmServiceProvider.java

```diff
@@ -0,0 +1,29 @@
+package com.comet.opik.infrastructure.llm.gemini;
+
+import com.comet.opik.api.AutomationRuleEvaluatorLlmAsJudge;
+import com.comet.opik.api.LlmProvider;
+import com.comet.opik.domain.llm.LlmProviderFactory;
+import com.comet.opik.domain.llm.LlmProviderService;
+import com.comet.opik.infrastructure.llm.LlmServiceProvider;
+import dev.langchain4j.model.chat.ChatLanguageModel;
+
+public class GeminiLlmServiceProvider implements LlmServiceProvider {
+
+    private final GeminiClientGenerator clientGenerator;
+
+    GeminiLlmServiceProvider(GeminiClientGenerator clientGenerator, LlmProviderFactory factory) {
+        this.clientGenerator = clientGenerator;
+        factory.register(LlmProvider.GEMINI, this);
+    }
+
+    @Override
+    public LlmProviderService getService(String apiKey) {
```

Comment:
```suggestion
    public LlmProviderService getService(@NonNull String apiKey) {
```
Other functions in this class as well